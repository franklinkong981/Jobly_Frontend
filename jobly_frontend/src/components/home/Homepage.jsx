import React, {useContext} from "react";

import {Button} from "reactstrap";

function Homepage() {
  const {currentUserInfo} = useContext(CountContext);

  return (
    <div className="Homepage">
      <h1>{currentUserInfo ? `Welcome back to Jobly, ${currentUserInfo.username}` : "Welcome to Jobly!"}</h1>
      <h2>Here you can find companies and job openings all in one convenient place.</h2>
      {currentUserInfo || (
        <div className="Homepage-buttons">
          <Link to={`/signup`} key={`homepage-signup-link`}>
            <Button color="primary">Log In</Button>
          </Link>
          <Link to={`/login`} key={`homepage-login-link`}>
            <Button color="secondary">Create An Account</Button>
          </Link>
        </div>
      )}
    </div>
  );
}

export default Homepage;