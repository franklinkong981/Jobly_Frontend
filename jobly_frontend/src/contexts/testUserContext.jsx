import React from "react";
import CurrentUserContext from "./currentUserContext.jsx";

const testUser = {
  username: "testuser",
  first_name: "Testy",
  last_name: "Usery",
  email: "testuser981@gmail.com"
};

const TestUserProvider = ({children, currentTestUser = testUser, hasUserAppliedToJob= () => false}) => (
  <CurrentUserContext.Provider value={{currentTestUser, hasUserAppliedToJob}}>
    {children}
  </CurrentUserContext.Provider>
);

export {TestUserProvider};