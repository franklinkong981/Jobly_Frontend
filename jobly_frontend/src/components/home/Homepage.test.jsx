import React from "react";
import {render} from "@testing-library/react";
import {MemoryRouter} from "react-router-dom";

import Homepage from "./Homepage.jsx";

import {TestUserProvider} from "../../contexts/testUserContext.jsx";
import { expect, it } from "vitest";

it("renders without crashing", function() {
  render(
    <MemoryRouter>
      <TestUserProvider>
        <Homepage />
      </TestUserProvider>
    </MemoryRouter>
  )
});

it("matches snapshot when logged in", function() {
  const {asFragment} = render(
    <MemoryRouter>
      <TestUserProvider>
        <Homepage/>
      </TestUserProvider>
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});

it("matches snapshot when logged out", function() {
  const {asFragment} = render(
    <MemoryRouter>
      <TestUserProvider currentTestUser={null}>
        <Homepage/>
      </TestUserProvider>
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});