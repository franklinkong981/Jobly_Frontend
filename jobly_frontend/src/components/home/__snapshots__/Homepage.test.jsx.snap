// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`matches snapshot when logged in 1`] = `
<DocumentFragment>
  <div
    class="Homepage"
  >
    <div
      class="Homepage-logged-out container text-center"
    >
      <h1
        class="Homepage-logged-out-title mb-4 font-weight-bold"
      >
        Welcome to Jobly!
      </h1>
      <p
        class="Homepage-logged-out-subtitle lead"
      >
        Find all companies and job openings here in one convenient place.
      </p>
      <div
        class="Homepage-logged-out-links"
      >
        <a
          class="Homepage-logged-out-login-link btn btn-secondary font-weight-bold mr-3"
          data-discover="true"
          href="/login"
        >
          Log In
        </a>
        <a
          class="Homepage-logged-out-signup-link btn btn-info font-weight-bold"
          data-discover="true"
          href="/signup"
        >
          Create Account
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`matches snapshot when logged out 1`] = `
<DocumentFragment>
  <div
    class="Homepage"
  >
    <div
      class="Homepage-logged-out container text-center"
    >
      <h1
        class="Homepage-logged-out-title mb-4 font-weight-bold"
      >
        Welcome to Jobly!
      </h1>
      <p
        class="Homepage-logged-out-subtitle lead"
      >
        Find all companies and job openings here in one convenient place.
      </p>
      <div
        class="Homepage-logged-out-links"
      >
        <a
          class="Homepage-logged-out-login-link btn btn-secondary font-weight-bold mr-3"
          data-discover="true"
          href="/login"
        >
          Log In
        </a>
        <a
          class="Homepage-logged-out-signup-link btn btn-info font-weight-bold"
          data-discover="true"
          href="/signup"
        >
          Create Account
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
